jsonargparse_tests/test_dataclass_like.py

```diff
@@ -978,3 +984,17 @@ def test_field_init_false(self, parser):
         assert cfg == Namespace()
         init = parser.instantiate_classes(cfg)
         assert init.data.p1 == {}
+
+    def test_nested_dataclass_with_default(self, parser):
+        # prior to changes in `jsonargparse._signatures.dataclass_to_dict` adding the nested
+        # attrs dataclass would lead to a TypeError with trying to use `dataclasses.asdict`
+        # on an attrs dataclass
```

Comment:
```suggestion
```

There is no reason to describe how it incorrectly worked before. I know there is a test above with such a comment, for which the same applies and I will probably remove later on.